### PR TITLE
SDPSubarray Configure and ConfigureScan write to config-db [ORCA-92]

### PR DESCRIPTION
**Pull request by brianmcilwrath**
Use of the 'ska_sdp_config' module added to the SDPSubarray Configure() command and appears to work. Still need to inspect that the database contains an entry with all expected and required fields!

ConfigureScan() discussed with Ben and it looked as if we need to discuss/think further on requirements for this function

### CODE REVIEW

File: src/tango_sdp_subarray/SDPSubarray/SDPSubarray.py [hunk 1]
```diff
@@ -230,6 +232,16 @@ def Configure(self, pb_config, schema_path=None):
 
         pb_config = json.loads(pb_config)
         validate(pb_config, schema)
+
+        for txn in db_config.Config().txn():
+            confdata = pb_config['configure']
+            pb_id = txn.new_processing_block_id(confdata['id'])
```
Comment (scpmw) [inline src/tango_sdp_subarray/SDPSubarray/SDPSubarray.py]: The "type" argument to `new_processing_block_id` should be either "batch" or "realtime". The two options here are really:
1. Just hardcode "realtime" for the moment
2. Don't bother with calling `new_processing_block_id` at all and just use `confdata['id']` for `pb_id` (that's what I expected!). Disadvantage here is that things will go wrong if Configure gets called again with the same id (which - to be fair - should be an error anyway).

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -21,7 +21,17 @@ stages:
...
+  before_script:
+    - apt update
+    - apt install -y wget
+    - pip install ska-sdp-config
   script:
+    - cd /usr/local/src
+    - wget "https://github.com/coreos/etcd/releases/download/v3.3.9/etcd-v3.3.9-linux-amd64.tar.gz"
+    - tar -xvf etcd-v3.3.9-linux-amd64.tar.gz
+    - mv etcd-v3.3.9-linux-amd64/etcd* /usr/local/bin/
+    - etcd &
```
Comment (scpmw) [inline .gitlab-ci.yml]: See the solution on `config_db_tests`:
```
    services:
      - name: quay.io/coreos/etcd:latest
        alias: etcd
        command:
          - /usr/local/bin/etcd
          - "--advertise-client-urls=http://0.0.0.0:2379"
          - "--listen-client-urls=http://0.0.0.0:2379"
          - "--initial-advertise-peer-urls=http://0.0.0.0:2380"
          - "--listen-peer-urls=http://0.0.0.0:2380"
          - "--initial-cluster=default=http://0.0.0.0:2380"
    variables:
      SDP_TEST_HOST: etcd
```

It's a bit wordy mainly because etcd configuration is a bit all-or-nothing.

File: src/tango_sdp_subarray/SDPSubarray/SDPSubarray.py [hunk 3]
```diff
@@ -230,6 +232,16 @@ def Configure(self, pb_config, schema_path=None):
 
         pb_config = json.loads(pb_config)
         validate(pb_config, schema)
+
+        for txn in db_config.Config().txn():
```
[the remark this answers is not in the document]
Comment (brianmcilwrath) [reply src/tango_sdp_subarray/SDPSubarray/SDPSubarray.py]: I don't think that storing a variable within a Tango device server is a good practice. Perhaps I should just close the client each time

**Comment by brianmcilwrath** [comment]
I have made the etcd3 connection a global variable and initialised it in the 'init_device()' method. I have also tried to incorporate Peter's comment on the pb_id use!
